ICRCs/ICRC-106/ICRC-106.md

```diff
@@ -0,0 +1,210 @@
+| Status |
+|:------:|
+|Draft|
+
+# ICRC-106: Standard for Associating Index Canisters with ICRC-1 Tokens
+
+## 1. Introduction
+
+Wallet applications and token management tools often need to retrieve both token metadata and transaction history for a given principal. However, identifying an associated index canister for ICRC-1 tokens is currently unstandardized, leading to inconsistencies in wallet integrations.
+
+**ICRC-106** introduces a standard approach for:
+1. Indicating the presence of an index canister for ICRC-1 tokens through ledger metadata.
+2. Defining a minimal interface for the index canister to facilitate querying transaction history in a consistent manner.
+
```

Comment:
```suggestion

**Which canisters are in scope of this standard?** 

The standard is specifying the behavior of an Index canister. Additionally, it suggests, but does not prescribe, how to design the API of the corresponding Ledger canister.
```

Reply:
This is more accurate:

Standard **ICRC-106** :
1. Introduces a standard approach for indicating the presence of an index canister for ICRC-1 tokens through ledger metadata.
2. Defines a minimal interface for the associated index canister to facilitate querying transaction history in a consistent manner.

Since the index is not standardized anywhere, here we prescribe the API one should expect, since otherwise discovering the index is not very useful

Reply:
Thanks, this is better. 

I suppose it's still a bit implicit what canisters are restricted by this standard. 

To make it concrete, which of the following statements would make sense:
1. Our Index complies with ICRC-106, but our Ledger does not support that standard yet.
2. Our Ledger complies with ICRC-106, but our Index does not support that standard yet.
3. We have an ICRC-106-compliant Ledger suite.